spec/services/builders/action_handlers/repository_spec.rb

```diff
@@ -0,0 +1,35 @@
+require 'rails_helper'
+
+describe ActionHandlers::Repository do
+  describe '.call' do
+    let(:payload) { create(:repository_event_payload, action: action) }
+    let!(:repository) { create(:repository) }
+
+    subject(:handle_action) { described_class.call(payload: payload, entity: repository) }
+
+    handleable_actions = %w[deleted archived transferred].freeze
+    unhandleable_actions = Events::Repository.actions.values - handleable_actions
+
+    context 'when the action is handleable' do
+      handleable_actions.each do |action|
+        let(:action) { action }
+
+        it 'marks the project as deleted' do
+          expect { handle_action }.to change(Project, :count)
+          expect(repository.project.deleted?).to be(true)
```

Comment:
I recommend having one assertion per example and place the `each` block inside each example.